### ISSUE TITLE
New icons

### ISSUE THREAD

**Issue by kkapsner** [kkapsner/CanvasBlocker#127]
As suggested by @JGH1000 in #112 a nicer addon icon would be good.

The icon should be 64x64 in size and a PNG.

I quite like the first suggestions from @Thorin-Oakenpants - I would stick with the black/gray-red colour scheme. The idea with the footprint has a nice twist maybe you can go further from that.

PS: while we are on it we should also do nice icons for the new page action (https://developer.mozilla.org/en-US/Add-ons/WebExtensions/user_interface/Page_actions) that will replace the notification in the WebExtension - neede sizes: 19x19 and 38x38). Not sure if it should be the same or something different that indicates that something was faked.

**Comment by Thorin-Oakenpants** [comment]
I will have a play in the next couple of days (tis the weekend, so beer and sports and stuff!). How quick do you need this stuff?

Just think about the color. RED is horrible, red is the nastiest color imaginable, even codecs hate it (lol). Green is friendly, environmental, safe, it even means go at traffic lights. An X means blocked. So a **_Green X_** = **_safely blocked_**. But the color is trivial, we can easily change that after we get the design right.

I have some ideas, and can easily give you a dozen to play with -i.e test how they look in about:addons, on the toolbar, in a dropdown UI hanger, in compact mode on toolbar, in dark theme. After you chose one we can tweak the color.

Question: I assume their will be a toolbar button available? I only ask because there are considerations for it looking OK in compact mode.

**Comment by kkapsner** [comment]
For the icon I have no deadline - I can change this whenever I want to.

I currently do not have a toobar button. Not sure what functionality I should put behind it. But if you work with svg we can just use that there. The smallest display used is 16x16. So it would be perfect if the icon would look good in 16x16, 32x32 and 64x64.

For the page action: the icon is only displayed when something was faked. So there we do not need a difference. If wee just use the normal icon (maybe without the X) this should be good - I think.

**Comment by kkapsner** [comment]
I added you as a collaborator so you can directly upload your suggestions to the repository.

You can put your suggestions into https://github.com/kkapsner/CanvasBlocker/tree/master/icons/.suggestions

**Comment by Thorin-Oakenpants** [comment]
FYI: fingerprint is from [here](https://openclipart.org/detail/278214/fingerprint) by TikiGawd .. it's a bit "fat" (less gap, more print).

Every time I worked this, the problem arose that any symbol had to be small enough, or off center enough as to not obscure the "print" (otherwise it's not easy to tell what it represents - as soon as the center of the print gets covered, it's a mess). And the smaller the icon, the harder it is to make things out - so the bigger the X the better, right? (*forget about 16px toolbar compact mode - everything looks a bit rough in that size, just as long as it's recognizable).

Came up with this idea - incorporate the X into the print.
![cb-06-combined](https://user-images.githubusercontent.com/16656956/30538712-59e8a6c8-9cc3-11e7-9066-b84feb819bc9.png)

counting images as
1, 2, 3
4, 5, 6

Note: background is off white, print is off-black, red can be tweaked to any shade - the "brighter" the more it stands out, but too bright and it looks cr*p

- 1 & 2 are the base images with either black or red edging. The corners are a simplistic presentation of "film" stock (eg kodak). I could change this to a two full edge strips with holes in them for feeding film, but then it eats space. You can mix these up, eg black edge, red corners, but it looks too funky!
- 3: I just used type to create a X and converted it to an object for resizing, then rastered it and then removed the parts not overlaying the print. I'm only showing one attempt at an X (I tried quite a few fonts and thicknesses and distortions). Its tricky, too small/thin it doesn't show or stand out as we shrink down, too fat/big and it engulfs the whole print
- 4: to make it stand out a fraction more, I added some gap around the edge of the x (same color as the background)
- 5: probably overkill, but I also added a slight stroking of the X on the inside (black edging)
- 6: same as 5 but red edge/corners

Told you I only dabbled. I have this as a PSD (2500px square) if anyone wants it.

@kkapsner - do you want me to convert a a couple of the above to 64png for you to play with?

**Comment by Thorin-Oakenpants** [comment]
A thinner X looks better, and the corners are too distracting. Color needs fixing its looks too orange against grey. The print can be lightened. The cutout of the x isn't really working
![cb-10-simple](https://user-images.githubusercontent.com/16656956/30562681-77e725be-9d13-11e7-8970-028cd453fd63.png)

**Comment by kkapsner** [comment]
Your suggestions look quite good. Can make a version of the last one (with the gradient) with rounded corners?

**Comment by Thorin-Oakenpants** [comment]
I've downloaded [Inkscape](https://portableapps.com/apps/graphics_pictures/inkscape_portable) - portable windows vector software, and am having a play. This way at least anything I spit out is a proper SVG, and thus you won;t get cr**py results when downsized. Don't expect anything in a hurry

By rounded corners, you mean like the green one I posted in the other thread? : https://github.com/kkapsner/CanvasBlocker/issues/112#issuecomment-329914248

**Comment by Thorin-Oakenpants** [comment]
Can't upload to that folder (tried on a nilla profile). I have 4 svgs for you to look at. I even zipped them. No dice... it just idles on `uploading X of Y` and never finishes

**Comment by kkapsner** [comment]
Yes, I mean like n the other thread.

I have no idea why you cannot upload - frankly I never use the web interface to upload anything. I just push my local repository...

**Comment by kkapsner** [comment]
I played a little bit with Inkscape and uploaded some svgs. I think the gradient one looks quite good also when displayed in small: 
![grafik](https://user-images.githubusercontent.com/1105157/30721963-d53bc0a4-9f2e-11e7-8c1e-0d6cdaaeb200.png)

**Comment by kkapsner** [comment]
I really like this icon - what are your opinions? @Thorin-Oakenpants? @JGH1000? @spodermenpls?

**Comment by spodermenpls** [comment]
@kkapsner The new icon is delightful - and gets displayed now instead of the green placeholder.

**Comment by Thorin-Oakenpants** [comment]
As long as it stands out and is different, works for me :)

I actually have a svg here (that was what I was trying to upload days ago) which is a _really really simple_ print (in black or grey or white or whatever) and in front of it is a shield (whatever color outliine we want, probably the same as the print) - the whole this is transparent background. We could color the shield in green or red or leave transparent. And thus you could have a dark and light themed icon set (or choose grey to cover both)

**Comment by Thorin-Oakenpants** [comment]
^^ But yeah - it looks good - here is someone's pic linked on [this reddit thread](https://www.reddit.com/r/firefox/comments/7233fp/canvasblocker_beta_webextension_released/)

https://i.imgur.com/bByfrQE.png

**Comment by JGH1000** [comment]
Great icon @kkapsner, the only small tweak I would do is to make the black border a bit thinner (would make the icon a bit cleaner). Otherwise looks really good!

**Comment by kkapsner** [comment]
Good point. Made it thinner: 
![grafik](https://user-images.githubusercontent.com/1105157/30785345-073c32ee-a166-11e7-9633-28d99034a01b.png) 
![grafik](https://user-images.githubusercontent.com/1105157/30785350-102d3862-a166-11e7-826c-8a22bcf62e60.png)

**Comment by kkapsner** [comment]
Then let's use this icon.

**Comment by kkapsner** [comment]
@Thorin-Oakenpants: I removed you as a collaborator - or did you want to stay it?

**Comment by Thorin-Oakenpants** [comment]
nah, she's cool :) Thanks for the WE landing on AMO 💋 . I updated (default basic settings) and bam, can't see the emojis on github - but I can see them here - https://www.webpagefx.com/tools/emoji-cheat-sheet/

I disabled CB, restarted, cleared everything .. same issue. uMatrix, uBlock Origin show some github assets cdn being blocked. **Weird**. I still get a symbol though, a basic black and white outline. Haven;t tried any other sites that use them - just wondering if you had come across this

I have a copy of my profile from 12 hrs ago, so I can load a copy of that and play around. I'm just going to assume its because of legacy issues - non e01s, legacy extensions

**Comment by Thorin-Oakenpants** [comment]
OK, should I open a new issue - something is definitely tied to CB. I loaded the console, cleared it, and refreshed this page as it has a :kiss: (emoji thing) in it .. and I got a console error (repeated 5 times - maybe there are 5 different emoji's on this page). Repeated with CB disabled and no console error.

[will post the error in a sec]

**Comment by kkapsner** [comment]
Pease open a new issue when you have done your tests and think it's tied to CB.
I do not have any problems with this page and do not get any errors.

**Comment by Thorin-Oakenpants** [comment]
Well that's good to know **that its just me** (ain't I the special one!), which is what I thought - just thought I'd ask :)

**Comment by kkapsner** [comment]
Better ask once too much than once too less.

**Comment by Thorin-Oakenpants** [comment]
Without going down a rabbit hole, the same sort of obscure weird behaviors happen with uBlock Origin WE (and still do if I allow some Greasemonkey/Violentmonkey scripts). All these things (stand-alone images not loading, scripts breaking, this emoji thing here) are ***all symptoms*** of a larger picture ...

... I am not e10s yet (1 extension holding me back), and I am not fully WE (waiting on NoScript and AMO reviews)... once I am e10s and have zero legacy extensions, I am sure everything will be fine

**Comment by kkapsner** [comment]
If you get better insights of the larger picture and think CB is part of the problem - just open an issue with a reproduction scenario.
